### PR TITLE
Health workflow updates

### PR DESCRIPTION
**Pull request by mosuem**
- Group unlicensed files into touched/not touched in PR, to avoid files without licenses already existing in the repo to fail the workflow.
- Add a flag to opt-out of running web tests for coverage.
- Look only at packages with changelogs for the changelog check.

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: .github/workflows/health.yaml [hunk 1]
```diff
@@ -50,6 +50,16 @@ on:
...
         type: boolean
         required: false
+      upload_coverage:
+        description: Whether to upload the coverage to coveralls
+        default: true
+        type: boolean
+        required: false
+      test_web:
+        description: Whether to run `dart test -p chrome` for coverage
+        default: true
```
Comment (devoncarew) [inline .github/workflows/health.yaml:60]: perhaps `default: false`?

File: .github/workflows/health.yaml [hunk 2]
```diff
@@ -50,6 +50,16 @@ on:
         default: false
         type: boolean
         required: false
+      upload_coverage:
+        description: Whether to upload the coverage to coveralls
+        default: true
+        type: boolean
+        required: false
+      test_web:
```
Comment (devoncarew) [inline .github/workflows/health.yaml:58]: `test_web`? `coverage_web`?

File: pkgs/firehose/lib/src/utils.dart [hunk 3]
```diff
@@ -4,6 +4,7 @@
 
 import 'dart:convert';
 import 'dart:io';
+import 'package:path/path.dart' as path;
```
Comment (devoncarew) [inline pkgs/firehose/lib/src/utils.dart:7]: nit: run `Organize imports` on this file

File: pkgs/firehose/lib/src/utils.dart [hunk 4]
```diff
@@ -109,6 +110,9 @@ bool expectEnv(String? value, String name) {
   }
 }
 
+String toRelative(File file) =>
+    path.relative(file.path, from: Directory.current.path);
```
Comment (devoncarew) [inline pkgs/firehose/lib/src/utils.dart:114]: Could also consider:

```
extension FileExtension on File {
  String get relative => p.relative(path, from: Directory.current.path);
}
```

**Review by devoncarew — APPROVED** [review]

File: .github/workflows/health.yaml (hunk 1 above)
Comment (mosuem) [reply .github/workflows/health.yaml:60]: I was thinking about that, not sure what the right default is. I can try `false`.

**Pull request merged**